src/Features/Core/Portable/BraceCompletion/IBraceCompletionService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.BraceCompletion
+{
+    internal interface IBraceCompletionService : ILanguageService
+    {
+        /// <summary>
+        /// Checks if this brace completion service should be the one used to provide brace completions at
+        /// the specified position with the specified opening brace.
+        /// Note that this does not mean that the service will provide brace completion, but that
+        /// only this service will be asked to provide them.
+        /// </summary>
+        Task<bool> IsValidForBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken);
```

Comment:
CanProvideBraceCompletion.

Reply:
can you doc the args?  from looking at it, it seems redundant.  i.e. Doc+OpenPosition = brace.